product/models/products/product.py

```diff
@@ -0,0 +1,23 @@
+from datetime import datetime
+from typing import Annotated, Literal
+
+from pydantic import BaseModel, Field, PositiveInt
+from pydantic.functional_validators import AfterValidator
+
+from product.models.products.validators import validate_product_id
+
+ProductId = Annotated[str, Field(min_length=36, max_length=36), AfterValidator(validate_product_id)]
+
+
+class Product(BaseModel):
+    name: Annotated[str, Field(min_length=1, max_length=30)]
+    id: ProductId
+    price: PositiveInt
+
+
+class ProductChangeNotification(BaseModel):
```

Comment:
not critical - but shouldnt we also send here the actual product entry and not just the id? 
i mean the name and price? whoever is listening for the event, could use this data instead of calling the get product api again, no?